src/arrayMethodJoin.js

```diff
@@ -4,8 +4,14 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedStr = this[0] === null || this[0] === undefined ? '' : `${this[0]}`;// eslint-disable-line
+
+    for (let i = 1; i < this.length; i++) {
+      joinedStr += `${separator}` + (this[i] === null || this[i] === undefined ? '' : `${this[i]}`);// eslint-disable-line
```

Comment:
I don't know if we should use the (ternary) operator...
but for me this code like difficult for reading.